sugarscape.py

```diff
@@ -960,30 +963,42 @@ def verifyConfiguration(configuration):
         configuration["startingAgents"] = totalCells
 
     # Ensure infinitely-lived agents are properly initialized
-    if configuration["agentMaxAge"][0] < 0 or configuration["agentMaxAge"][1] < 0:
-        configuration["agentMaxAge"][0] = -1
-        configuration["agentMaxAge"][1] = -1
+    if configuration["agentMaxAge"][0] < 0:
+        configuration["agentMaxAge"] = [-1, -1]
+
+    if configuration["agentSelfishnessFactor"][0] < 0:  
+        configuration["agentSelfishnessFactor"] = [-1, -1]
+    elif configuration["agentSelfishnessFactor"][1] > 1:
+        configuration["agentSelfishnessFactor"][1] = 1
+   
+    if configuration["agentDecisionModelTribalFactor"][0] < 0:
+        configuration["agentDecisionModelTribalFactor"] = [-1, -1]
+    elif configuration["agentDecisionModelTribalFactor"][1] > 1:
+        configuration["agentDecisionModelTribalFactor"][1] = 1
 
     if configuration["agentTagStringLength"] < 0:
         configuration["agentTagStringLength"] = 0
     if configuration["environmentMaxTribes"] < 0:
         configuration["environmentMaxTribes"] = 0
+    
     # Ensure at most number of tribes is equal to agent tag string length
     if configuration["agentTagStringLength"] > 0 and configuration["environmentMaxTribes"] > configuration["agentTagStringLength"]:
         configuration["environmentMaxTribes"] = configuration["agentTagStringLength"]
 
-    configuration["agentDecisionModelTribalFactor"].sort()
-    if configuration["agentDecisionModelTribalFactor"][1] > 1:
-        configuration["agentDecisionModelTribalFactor"][1] = 1
-    elif configuration["agentDecisionModelTribalFactor"][0] < 0:
-        configuration["agentDecisionModelTribalFactor"][0] = -1
-
-    # Ensure at most number of tribes is equal to the number of colors in the GUI
-    maxTribes = 25
-    if configuration["environmentMaxTribes"] > maxTribes:
+    # Ensure at most number of tribes and ethical decisions are equal to the number of colors in the GUI
```

Comment:
We're calling these decision models rather than ethical decision-whatever so we free ourselves up to also do social science work.

Reply:
Done.